ui/jstests/test_manage_taxonomies.jsx

```diff
@@ -35,39 +35,278 @@ define(['QUnit', 'jquery', 'setup_manage_taxonomies', 'reactaddons',
-        responseText: addTermResponse,
+        url: "/api/v1/repositories/repo/vocabularies/easy/terms/",
+        type: "POST",
+        status: 400
+      });
+      TestUtils.initMockjax({
+        url: "/api/v1/repositories/repo/vocabularies/difficulty/terms/",
+        responseText: term,
+        type: "POST"
+      });
+      TestUtils.initMockjax({
+        url: "/api/v1/repositories/repo/vocabularies/difficulty/terms/",
+        responseText: listOfTerms,
+        type: "GET"
+      });
+      TestUtils.initMockjax({
+        url: "/api/v1/repositories/repo/vocabularies/",
+        contentType: "application/json; charset=utf-8",
+        responseText: vocabulary,
+        dataType: 'json',
         type: "POST"
       });
+      TestUtils.initMockjax({
+        url: "/api/v1/repositories/repo2/vocabularies/",
+        type: "POST",
+        status: 400
+      });
+      TestUtils.initMockjax({
+        url: "/api/v1/repositories/repo/vocabularies/",
+        contentType: "application/json; charset=utf-8",
+        responseText: listOfVocabularies,
+        dataType: 'json',
+        type: "GET"
+      });
+      TestUtils.initMockjax({
+        url: "/api/v1/learning_resource_types/",
+        contentType: "application/json; charset=utf-8",
+        responseText: learningResourceTypes,
+        dataType: 'json',
+        type: "GET"
+      });
     },
     afterEach: function() {
       TestUtils.cleanup();
     }
   });
 
+  QUnit.test('Assert that TermComponent renders properly',
+    function(assert) {
+      assert.ok(TermComponent, "class object not found");
+      var term = {
+        "id": 9,
+        "slug": "test",
+        "label": "test",
+        "weight": 1
+      };
+      var termComponentRendered = React.addons.TestUtils.
+        renderIntoDocument(
+          <TermComponent
+            term={term}
+          />
+        );
+
+      var labelComponent = React.addons.TestUtils.
+        findRenderedDOMComponentWithTag(
+          termComponentRendered,
+          'label'
+      );
+      //testing label value render
+      var label = labelComponent.getDOMNode();
+      assert.equal(label.innerHTML, term.label);
+    }
+  );
+
   QUnit.test('Assert that VocabularyComponent renders properly',
     function(assert) {
       assert.ok(VocabularyComponent, "class object not found");
-
       var done = assert.async();
 
       var addTermCalled = 0;
       var addTerm = function() {
         addTermCalled += 1;
       };
-
+      var reportError = function() {};
       var afterMount = function(component) {
         var node = React.findDOMNode(component);
         var textbox = $(node).find("input")[0];
+
+        // check term title
+        var listOfLinks = React.addons.TestUtils.
+          scryRenderedDOMComponentsWithTag(
+            component,
+            'a'
+        );
+        assert.equal(listOfLinks.length, 2);
+        var linkHeader = listOfLinks[0].getDOMNode();
```

Comment:
Change this to `React.findDOMNode(listOfLinks[0])` since `getDOMNode()` is deprecated